### PR TITLE
Generator

### PR DESCRIPTION
**Pull request by harimohanraj89**
**Story card:** [sc-7970](https://app.shortcut.com/simpledotorg/story/7970/create-generator-for-new-deployment)

This PR adds a `bin/new` generator script to the standalone repo. Given a deployment name and subdomain, it can create the necessary files to get a new instance up and running.

### Question

Do you think we should collect the server IPs in this script as well, wizard-ize the full creation of the inventory file?

**YES** I've updated the PR to collect IPs (or leave them blank and configure later) for maximum convenience.

### Sample

```
Enter the name of the new deployment.
The deployment name can only contain letters and underscores, eg: 'test_environment'.
Deployment name: test

Enter the domain or subdomain on which you'd like to host your Simple server instance
Eg. api.simple.org
Domain/Subdomain: test

Enter the public IP addresses of the servers or VMs that will host the following services.
If you don't have the IP addresses yet, you can leave these fields blank, and configure them later.
Primary postgres database: 1.1.1.1
Secondary postgres database (hot standby follower): 1.1.1.2
Web servers (comma-separated list): 1.1.1.3
Sidekiq background job servers (comma-separated list): 1.1.1.4,1.1.1.5
Primary Redis cache: 1.1.1.6
Secondary Redis cache (hot standby follower): 1.1.1.7
Prometheus (system monitoring tool): 1.1.1.8
Grafana (dashboards for viewing Prometheus data): 1.1.1.9
Storage of application logs, audit logs, and database backups: 2.2.2.2
Load balancer (this server will be the entry point for all incoming web traffic): 2.2.2.5

Would you like to generate an Ansible vault password for this deployment? Select No if you already have an Ansible
vault password that you plan to use
Generate Ansible vault password? (y/N): y

Created /Users/hariharanmohanraj/code/simple/simple-standalone/hosts/test...
Created /Users/hariharanmohanraj/code/simple/simple-standalone/group_vars/test/vars.yml...
Created /Users/hariharanmohanraj/code/simple/simple-standalone/group_vars/test/vault.yml.decrypted...
Updated /Users/hariharanmohanraj/code/simple/simple-standalone/ansible.cfg...
Created /Users/hariharanmohanraj/.vault_password_test...

test deployment environment initialized!
To finish setup, follow these steps
- If you skipped entering any IP addresses, enter them into your Ansible inventory file hosts/test
- Configure your instance's variables in group_vars/test/vars.yml
- Configure your instance's secrets in group_vars/test/vault.yml.decrypted
- Create your vault file by running `make encrypt hosts=test`

And you're done! You can then deploy Simple onto your servers with `make all hosts=test`
```

### CODE REVIEW

File: bin/new [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# Set up deployment files
+# ----------------------------
+project_path = File.expand_path("..", File.dirname(__FILE__))
+templates_path = File.expand_path("bin/templates", project_path)
+hosts_path = File.expand_path("hosts", project_path)
+group_vars_path = File.expand_path("group_vars/#{deployment_name}", project_path)
+
+# Ansible inventory file
+inventory_template = ERB.new(File.read("#{templates_path}/inventory.erb"))
+inventory = inventory_template.result(binding)
```
Comment (prabhanshuguptagit) [inline bin/new:64]: Is it possible to pass in the vars as a hash? Might be more amenable to writing some tests around this (if we get to it).

File: bin/new [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+# Print next steps
+# ----------------------------
+puts
+puts "#{deployment_name} deployment environment initialized!"
+puts "To finish setup, follow these steps"
+puts "- If you skipped entering any IP addresses, enter them into your Ansible inventory file hosts/#{deployment_name}"
+puts "- Configure your instance's variables in group_vars/#{deployment_name}/vars.yml"
+puts "- Configure your instance's secrets in group_vars/#{deployment_name}/vault.yml.decrypted"
+puts "- Create an Ansible vault password and place it in a file at ~/.vault_password_#{deployment_name}"
+puts "- Create your vault file by running `make encrypt hosts=#{deployment_name}`"
```
Comment (prabhanshuguptagit) [inline bin/new:105]: Should we accept any of these from the CLI as well? Ansible vault password?

**Review by prabhanshuguptagit — COMMENTED** [review]
Looks stellar, some minor usability comments.

**Review by prabhanshuguptagit — APPROVED** [review]

File: bin/new [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+# IP addresses
+puts
+puts "Enter the public IP addresses of the servers or VMs that will host the following services."
+puts "If you don't have the IP addresses yet, you can leave these fields blank, and configure them later."
+
+postgres_primary = input("Primary postgres database", default: "change-me")
+postgres_secondary = input("Secondary postgres database (hot standby follower)", default: "change-me")
+web_servers = input("Web servers (comma-separated)", default: "change-me,change-me").split(",").map!(&:strip)
+sidekiq_servers = input("Sidekiq background job servers (comma-separated)", default: "change-me,change-me").split(",").map!(&:strip)
```
[the remark this answers is not in the document]
Comment (harimohanraj89) [reply bin/new:47]: Great idea.

File: bin/new (hunk 1 above)
Comment (harimohanraj89) [reply bin/new:64]: Looks like we can use a `result_with_hash` method that'll take a hash. I'll do that. Passing a binding around is... fragile...

**Pull request merged**